PSReadLine/Suggestion.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+
+namespace Microsoft.PowerShell
+{
+    public partial class PSConsoleReadLine
+    {
+        private class SuggestionModeRestore : IDisposable
+        {
```

Comment:
See `UseViCommandModeTables` for an alternative pattern in use in this code base.

Reply:
Good to know this. I will switch to that pattern.